hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/common/SubmissionManager.java

```diff
@@ -0,0 +1,101 @@
+ */
+package com.hedera.node.app.workflows.common;
+
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.PLATFORM_TRANSACTION_NOT_CREATED;
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.node.app.SessionContext;
+import com.hedera.node.app.service.mono.records.RecordCache;
+import com.hedera.node.app.service.mono.stats.MiscSpeedometers;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import com.swirlds.common.system.Platform;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The {@code SubmissionManager} provides functionality to submit transactions to the platform. */
+public class SubmissionManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SubmissionManager.class);
+
+    private final Platform platform;
+    private final RecordCache recordCache;
+    private final MiscSpeedometers speedometers;
+
+    /**
+     * Constructor of {@code SubmissionManager}
+     *
+     * @param platform the {@link Platform}
+     * @param recordCache the {@link RecordCache}
+     * @param speedometers metrics related to submissions
+     */
+    public SubmissionManager(
+            @NonNull final Platform platform,
+            @NonNull final RecordCache recordCache,
+            @NonNull final MiscSpeedometers speedometers) {
+        this.platform = requireNonNull(platform);
+        this.recordCache = requireNonNull(recordCache);
+        this.speedometers = requireNonNull(speedometers);
+    }
+
+    /**
+     * Submit a transaction to the {@link Platform}
+     *
+     * @param ctx the {@link SessionContext}
+     * @param txBody the {@link TransactionBody} that should be submitted to the platform
+     * @param byteBuffer the {@link ByteBuffer} of the data that should be submitted
+     * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws PreCheckException if the transaction could not be submitted
+     */
+    public void submit(
+            @NonNull final SessionContext ctx,
+            @NonNull final TransactionBody txBody,
+            @NonNull final ByteBuffer byteBuffer)
+            throws PreCheckException {
+        requireNonNull(ctx);
+        requireNonNull(txBody);
+        requireNonNull(byteBuffer);
+
+        final byte[] payload;
+        if (txBody.hasUncheckedSubmit()) {
+            try {
+                payload =
+                        ctx.txParser()
+                                .parseFrom(txBody.getUncheckedSubmit().getTransactionBytes())
+                                .toByteArray();
+            } catch (InvalidProtocolBufferException e) {
+                LOG.warn("Transaction bytes from UncheckedSubmit not a valid gRPC transaction!", e);
+                throw new PreCheckException(PLATFORM_TRANSACTION_NOT_CREATED);
+            }
```

Comment:
What is an uncheckedSubmit? Maybe we should add some comment here to describe what an unchecked submit is. I didn't see it in the workflow diagram either. Does it belong in the SubmissionManager, or in the workflow? I'm not sure.

Also, we shouldn't use `LOG.warn` here, because any attacker could try to fill our logs just by submitting bogus transactions. We should reserve info, warn, and error/fatal for issues related to our code, not user submitted data, and use debug or trace for this kind of log statement.

Reply:
`uncheckedSubmit` is the privileged transaction in `NetworkService`.